src/CommunityToolkit.Maui.MediaElement/Extensions/SubtitleExtensions.macios.cs

```diff
@@ -0,0 +1,157 @@
+		this.player = player;
+		cues = [];
+		subtitleLabel = new UILabel
+		{
+			Frame = CalculateSubtitleFrame(playerViewController),
+			TextColor = UIColor.White,
+			TextAlignment = UITextAlignment.Center,
+			Font = UIFont.SystemFontOfSize(16),
+			Text = "",
+			Lines = 0,
+			AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleTopMargin | UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleBottomMargin
+		};
+		MediaManagerDelegate.WindowChanged += OnWindowStatusChanged;
+	}
+
+	/// <summary>
+	/// Loads the subtitles from the provided URL.
+	/// </summary>
+	/// <param name="mediaElement"></param>
+	public async Task LoadSubtitles(IMediaElement mediaElement)
+	{
+		this.mediaElement = mediaElement;
+		cues.Clear();
+		subtitleLabel.Font = UIFont.FromName(mediaElement.SubtitleFont, (float)mediaElement.SubtitleFontSize) ?? UIFont.SystemFontOfSize((float)mediaElement.SubtitleFontSize);
+		string? vttContent;
+		try
+		{
+			vttContent = await httpClient.GetStringAsync(mediaElement.SubtitleUrl);
+		}
+		catch (Exception ex)
+		{
+			System.Diagnostics.Trace.TraceError(ex.Message);
+			return;
+		}
+
+		cues = mediaElement.SubtitleUrl switch
+		{
+			var url when url.EndsWith("srt") => SrtParser.ParseSrtContent(vttContent),
+			var url when url.EndsWith("vtt") => VttParser.ParseVttContent(vttContent),
+			_ => throw new NotSupportedException("Unsupported subtitle format"),
+		};
+	}
+
+	/// <summary>
+	/// Starts the subtitle display.
+	/// </summary>
+	public void StartSubtitleDisplay()
+	{
+		ArgumentNullException.ThrowIfNull(subtitleLabel);
+		playerObserver = player?.AddPeriodicTimeObserver(CMTime.FromSeconds(1, 1), null, (time) =>
+		{
+			TimeSpan currentPlaybackTime = TimeSpan.FromSeconds(time.Seconds);
+			switch(viewController)
+			{
+				case null:
+					DispatchQueue.MainQueue.DispatchAsync(() => playerViewController.View?.AddSubview(subtitleLabel));
+					break;
+				default:
+					DispatchQueue.MainQueue.DispatchAsync(() => viewController?.View?.AddSubview(subtitleLabel));
+					break;
+			}
+			subtitleLabel.Frame = viewController is not null ? CalculateSubtitleFrame(viewController) : CalculateSubtitleFrame(playerViewController);
+			DispatchQueue.MainQueue.DispatchAsync(() => UpdateSubtitle(currentPlaybackTime));
+		});
+	}
+
+	/// <summary>
+	/// Stops the subtitle display.
+	/// </summary>
+	public void StopSubtitleDisplay()
+	{
+		if (playerObserver is not null)
+		{
+			player.RemoveTimeObserver(playerObserver);
+			playerObserver.Dispose();
+			playerObserver = null;
+			subtitleLabel.RemoveFromSuperview();
+		}
+	}
+	void UpdateSubtitle(TimeSpan currentPlaybackTime)
+	{
+		ArgumentNullException.ThrowIfNull(subtitleLabel);
+		foreach (var cue in cues)
+		{
+			if (currentPlaybackTime >= cue.StartTime && currentPlaybackTime <= cue.EndTime)
+			{
+				subtitleLabel.Text = cue.Text;
+				subtitleLabel.BackgroundColor = UIColor.FromRGBA(0, 0, 0, 128);
```

Comment:
can we cache those colors? That way we will not create new one on each interaction, please also cache at line 125